pkg/storegateway/bucket_store_server_test.go

```diff
@@ -163,6 +168,92 @@ func (s *storeTestServer) Series(ctx context.Context, req *storepb.SeriesRequest
 
 			seriesSet = append(seriesSet, copiedSeries)
 		}
+
+		if recvSeries := res.GetStreamingSeries(); recvSeries != nil {
+			if req.StreamingChunksBatchSize == 0 || req.SkipChunks {
+				err = errors.New("got a streaming series when streaming was disabled")
+				return
+			}
+
+			var recvSeriesData []byte
+
+			// We prefer to stay on the safest side at this stage
+			// so we do a marshal+unmarshal to copy the whole series.
+			recvSeriesData, err = recvSeries.Marshal()
+			if err != nil {
+				err = errors.Wrap(err, "marshal received series")
+				return
+			}
+
+			copiedSeries := &storepb.StreamSeriesBatch{}
+			if err = copiedSeries.Unmarshal(recvSeriesData); err != nil {
+				err = errors.Wrap(err, "unmarshal received series")
+				return
+			}
+
+			streamingSeriesSet = append(streamingSeriesSet, copiedSeries.Series...)
+
+			if recvSeries.IsEndOfSeriesStream {
+				break
+			}
+		}
+	}
+
+	if req.StreamingChunksBatchSize > 0 && !req.SkipChunks {
```

Comment:
can this be simplified to 
```suggestion
	if req.StreamingChunksBatchSize > 0 {
```

I see the same condition appearing in multiple places. If it makes sense, can you change them there too?